pallets/messages/src/rpc/src/tests/mod.rs

```diff
@@ -0,0 +1,117 @@
+mod rpc_mock;
+
+use super::*;
+use rpc_mock::*;
+
+use common_primitives::node::BlockNumber;
+use pallet_messages_runtime_api::MessagesApi as MessagesRuntimeApi;
+use std::sync::Arc;
+use substrate_test_runtime_client::runtime::Block;
+
+const SCHEMA_ID_EMPTY: u16 = 1;
+const SCHEMA_ID_HAS_MESSAGES: u16 = 2;
+
+fn test_messages() -> Vec<MessageResponse<BlockNumber>> {
+	vec![
+		MessageResponse {
+			payload: None,
+			msa_id: None,
+			provider_msa_id: 1,
+			index: 0,
+			block_number: 1,
+			cid: Some(vec![0, 1, 2, 3]),
+			payload_length: Some(42),
+		},
+		MessageResponse {
+			payload: None,
+			msa_id: None,
+			provider_msa_id: 1,
+			index: 1,
+			block_number: 1,
+			cid: Some(vec![0, 1, 2, 3]),
+			payload_length: Some(42),
+		},
+	]
+}
+
+sp_api::mock_impl_runtime_apis! {
+	impl MessagesRuntimeApi<Block, BlockNumber> for TestRuntimeApi {
+		fn get_schema_by_id(schema_id: SchemaId) -> Option<SchemaResponse> {
+			match schema_id {
+				SCHEMA_ID_EMPTY | SCHEMA_ID_HAS_MESSAGES => Some(SchemaResponse {
+					schema_id,
+					model: b"schema".to_vec(),
+					model_type: ModelType::AvroBinary,
+					payload_location: PayloadLocation::default(),
+				}),
+				_ => None,
+			}
+		}
+
+		fn get_messages_by_schema_and_block(schema_id: SchemaId, _schema_payload_location: PayloadLocation, _block_number: BlockNumber) ->
+			Vec<MessageResponse<BlockNumber>> {
+				match schema_id {
+					SCHEMA_ID_HAS_MESSAGES => test_messages(),
+					_ => vec![]
+				}
+			}
+	}
+}
+
+type GetMessagesBySchemaResult = Result<
+	common_primitives::messages::BlockPaginationResponse<
+		u32,
+		common_primitives::messages::MessageResponse<u32>,
+	>,
+	jsonrpsee::core::Error,
+>;
+
+#[tokio::test]
+async fn get_messages_by_schema_with_invalid_request_should_panic() {
```

Comment:
should there be a test where we properly ask for data, but which doesn't exist, such as a too-high block or index?

Reply:
I don't think that would test much with the mocks, but I ran a quick test locally. It just returns an empty array. Perhaps that is a test for a future e2e integration test.